Ballista Release Blog Announcement (#53)
apache/datafusion-site: content/blog/2025-02-02-datafusion-ballista-43.0.0.md

  ctx.register_csv("example", "tests/data/example.csv", CsvReadOptions::new()).await?;

  // create a plan to run a SQL query
  let df = ctx.sql("SELECT a, MIN(b) FROM example WHERE a <= b GROUP BY a LIMIT 100").await?;

  // execute and print results
  df.show().await?;
  Ok(())
}
```

Additionally, Ballista’s versioning scheme has been aligned with that of DataFusion, ensuring that Ballista's version number reflects the compatible DataFusion version..

### Removal of Experimental Features

Ballista had grown in scope to include several experimental features in various states of completeness. Some features have been removed from this release in an effort to strip Ballista back to its core and make it easier to maintain and extend.

Specifically, the caching subsystem, object store registry, plugin subsystem, key-value stores for persistent scheduler state, and the UI have been removed.

### Performance & Scalability

Ballista has significantly leveraged the advancements made in the DataFusion project over the past year. Benchmark results demonstrate notable improvements in performance, highlighting the impact of these enhancements:

Per query comparison:

<img

> src="https://github.com/apache/datafusion-ballista/raw/e9e8f9a783d9dbc8b40f3e57e2a7294788df6b93/docs/source/_static/images/tpch_queries_compare.png"

Reviewer: I wonder if it would be better to add these images as part of this PR rather than link to these GitHub URLs?

width="100%"
class="img-responsive"
alt="Per query comparison"
/>

Relative speedup:

<img
src="https://github.com/apache/datafusion-ballista/raw/e9e8f9a783d9dbc8b40f3e57e2a7294788df6b93/docs/source/_static/images/tpch_queries_speedup_rel.png"
width="100%"
class="img-responsive"
alt="Relative speedup graph"
/>

The overall speedup is 2.9x

<img
src="https://github.com/apache/datafusion-ballista/raw/e9e8f9a783d9dbc8b40f3e57e2a7294788df6b93/docs/source/_static/images/tpch_allqueries.png"
width="50%"
class="img-responsive"
alt="Overall speedup"
/>

### New Logo

> We got logo updated as well, visually similar to other DataFusion projects.

Reviewer: ```suggestion
Ballista now has a new logo, which is visually similar to other DataFusion projects.
```
